controllers/reconcile_reverse_proxy.go

```diff
@@ -148,18 +156,32 @@ func (r *ReverseProxyReconciliation) updateReverseProxyDeployment(deployment *ap
 		return err
 	}
 
+	// Get the desired volumes configuration
+	desiredVolumes := r.createReverseProxyVolumes()
+
 	// Get current container
 	currentContainer := &deployment.Spec.Template.Spec.Containers[0]
 
 	// Check if container needs update
-	needsUpdate, updateReason := r.compareContainer(currentContainer, &desiredContainer)
+	containerNeedsUpdate, updateReason := r.compareContainer(currentContainer, &desiredContainer)
 
-	if needsUpdate {
-		r.Log.Info("Updating reverse proxy deployment", "reason", updateReason)
+	// Check if volumes need update
+	volumesNeedsUpdate := r.compareVolumes(deployment.Spec.Template.Spec.Volumes, desiredVolumes)
+
+	if containerNeedsUpdate || volumesNeedsUpdate {
```

Comment:
I would like to revisit this comparison in the future - and if it's possible for us to reuse logic elsewhere to see if an update is necessary (I'm worried if we have a bug here we can either miss an update, or endlessly update in a loop).